Description: correctly count preprocessor options
 The dcc_expand_preprocessor_options function incorrectly handled
 several consecutive -Wp,... options. The problem was that even
 though it increased the number of arguments, it didn't increase
 argc, which could later be used as number of all arguments by
 the same code, hence deleting some of the last arguments.
Origin: other, http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=626926;msg=5
Bug: http://code.google.com/p/distcc/issues/detail?id=83
Bug-Debian: http://bugs.debian.org/626926
Bug-Gentoo: https://bugs.gentoo.org/351979
Applied-Upstream: 3.2, http://code.google.com/p/distcc/source/detail?r=750
Author: "Jiri Palecek" <jpalecek@web.de>

--- distcc-3.1.orig/source/src/arg.c
+++ distcc-3.1/source/src/arg.c
@@ -515,6 +515,8 @@
             }
             free(argv);
             *argv_ptr = argv = new_argv;
+            i += extra_args - 1;
+            argc += extra_args - 1;
         }
     }
     return 0;
